hudi-common/src/main/java/org/apache/hudi/common/util/CompactionUtils.java

```diff
@@ -214,22 +216,22 @@ public static List<HoodieInstant> getPendingCompactionInstantTimes(HoodieTableMe
    */
   public static Option<Pair<HoodieTimeline, HoodieInstant>> getDeltaCommitsSinceLatestCompaction(
       HoodieActiveTimeline activeTimeline) {
-    Option<HoodieInstant> lastCompaction = activeTimeline.getCommitTimeline()
+    Option<HoodieInstant> lastCompaction = activeTimeline.getCommitTimeline().filter(s -> !s.getAction().equals(REPLACE_COMMIT_ACTION))
         .filterCompletedInstants().lastInstant();
-    HoodieTimeline deltaCommits = activeTimeline.getDeltaCommitTimeline();
+    HoodieTimeline deltaAndReplaceCommits  = activeTimeline.getDeltaCommitAndReplaceCommitTimeline();
```

Comment:
but I am not sure if this makes sense. this method `getDeltaCommitsSinceLatestCompaction` only cares for delta commits for the purpose of scheduling compaction. So, replace commits does not matter. can you help me understand why we need to include repalce commits here.